dev/com.ibm.ws.logging.osgi/src/com/ibm/ws/logging/internal/osgi/stackjoiner/ThrowableProxyActivator.java

```diff
@@ -0,0 +1,396 @@
+     * component is ready for activation.
+     *
+     * @param bundleContext the bundleContext
+     */
+    protected void activate() throws Exception {    
+        // Store a reference to the printStackTraceOverride method from BaseTraceService
+        throwableInfo = new ThrowableInfo(inst);
+        
+        if (throwableInfo.isInitialized()) {
+        	String runtimeVersion = getRuntimeClassVersion();
+            if (runtimeVersion != null && !runtimeVersion.equals(getCurrentVersion())) {
+                // TODO: Use a compatibility check instead
+                throw new IllegalStateException("Incompatible proxy code (version " + runtimeVersion + ")");
+            }
+
+            // Find or create the proxy jar if the runtime code isn't loaded
+            if (runtimeVersion == null) {
+    	        JarFile proxyJar = getBootProxyJarIfCurrent();
+    	        if (proxyJar == null) {
+    	            proxyJar = createBootProxyJar();
+    	        }
+    	        inst.appendToBootstrapClassLoaderSearch(proxyJar);
+            }
+            
+        	activateThrowableProxyMethodTarget();
+    
+        	ThrowableClassFileTransformer tcfTransformer = new ThrowableClassFileTransformer();
+			//for (Class<?> clazz : ) {
+			  //  if (clazz.getName().equals("java.lang.Throwable")) {
+			    	try {
+			    		inst.addTransformer(tcfTransformer, true);
+			            inst.retransformClasses(Throwable.class); //something to replace retransfornm
+			        } catch (Throwable t) {
+			            t.printStackTrace();
+			        } finally {
+			        	inst.removeTransformer(tcfTransformer);
+			        }
+			   //  }
+	//		}
+        }
+    	
+	}
+	
+	protected void deactivate() throws Exception {
+        try {
+        	if (throwableInfo.isInitialized())
+        		deactivateThrowableProxyTarget();
+        } catch (Exception e) {
+            throw new Exception(e);
+        }
+    }
+	
+	/**
+     * Determine if the boot delegated proxy is already available and, if so,
+     * what its version is.
+     *
+     * @return the runtime version of the emitter proxy or null if the class
+     *         is not currently available
+     */
+    @FFDCIgnore(Exception.class)
+    String getRuntimeClassVersion() {
+        String runtimeVersion = null;
+        try {
+            Class<?> clazz = Class.forName(THROWABLE_PROXY_CLASS_NAME);
+            Field version = ReflectionHelper.getDeclaredField(clazz, VERSION_FIELD_NAME);
+            runtimeVersion = (String) version.get(null);
+        } catch (Exception e) {
+        }
+        return runtimeVersion;
+    }
+    
+	/**
+     * Create a jar file that contains the proxy code that will live in the
+     * boot delegation package.
+     *
+     * @return the jar file containing the proxy code to append to the boot
+     *         class path
+     *
+     * @throws IOException if a file I/O error occurs
+     */
+    JarFile createBootProxyJar() throws IOException {
```

Comment:
Currently, we are performing the jar creation at runtime because we need to use the ASM Framework to remap an existing proxy template class (within the current package) into a class that’s within the boot delegation package. Afterwards, we feed this newly created jar file to the bootstrap class loader. 
Since performing this at build-time will require much more investigation and since the additional time required to perform this a runtime is minimal ([Please see comment below](https://github.com/OpenLiberty/open-liberty/pull/15648#issuecomment-821446827)), we have decided to not create the jar at build-time .